mm2src/mm2_tests.rs

```diff
@@ -8414,6 +8414,42 @@ fn test_get_public_key() {
         "022cd3021a2197361fb70b862c412bc8e44cff6951fa1de45ceabfdd9b4c520420"
     )
 }
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn test_get_public_key_hash() {
+    let coins = json!([
+        {"coin":"RICK","asset":"RICK","rpcport":8923,"txversion":4,"protocol":{"type":"UTXO"}},
+    ]);
+
+    let mm = MarketMakerIt::start(
+        json!({
+            "gui": "nogui",
+            "netid": 9998,
+            "passphrase": "bob passphrase",
+            "rpc_password": "password",
+            "coins": coins,
+            "i_am_seed": true,
+        }),
+        "password".into(),
+        None,
+    )
+    .unwrap();
+    let (_dump_log, _dump_dashboard) = mm.mm_dump();
+    log!({"Log path: {}", mm.log_path.display()});
+    let resp = block_on(mm.rpc(json!({
+                 "userpass": "password",
+                 "mmrpc": "2.0",
+                 "method": "get_public_key_hash",
+                 "params": {},
+                 "id": 0})))
+    .unwrap();
+
+    // Must be 200
+    assert_eq!(resp.0, StatusCode::OK);
+    let v: RpcV2Response<GetPublicKeyHashResult> = serde_json::from_str(&*resp.1).unwrap();
+    // Public key hash must be "b506088aa2a3b4bb1da3a29bf00ce1a550ea1df9"
```

Comment:
Just interested to know how you got this public key hash to use on the test :)

Reply:
I can understand that the keypair is being generated using the passphrase in .env file. So confirming that, I ran the rpc method(get_public_key_hash) I created to get the public key hash to use for the test and I reran the test using the wrong passphrase to confirm a failed test and it worked just fine

Reply:
> So confirming that, I ran the rpc method(get_public_key_hash) I created to get the public key hash to use for the test

The problem with this approach is if you implemented the get_public_key_hash method wrongly you would still get a passable test. Unit tests have 2 purposes, the first is to check that your code works right which wouldn't work with this approach and the second purpose is to avoid regression when implementing new features.

The right approach if you checked this https://github.com/KomodoPlatform/atomicDEX-API/issues/1189#issue-1096360846 would have been to run mm2 using a certain seed and get the public key hash from the folder name under DB, then use the this seed and public key hash to test your new method.

Reply:
thanks noted.